src/main/java/Bebe.java

```diff
@@ -0,0 +1,330 @@
+        saveTasksToFile();
+    }
+
+    /**
+     * Lists all the tasks added by the user.
+     */
+    private static void listTasks() {
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < tasks.size(); i++) {
+            System.out.println((i + 1) + ". " + tasks.get(i).toString());
+        }
+    }
+
+    /**
+     * Marks a task as done.
+     *
+     * @param taskNumber The index of the task to mark as done.
+     */
+    private static void markTaskAsDone(String taskNumber) throws BebeException {
+        try {
+            int index = Integer.parseInt(taskNumber) - 1;
+            if (index >= 0 && index < tasks.size()) {
+                tasks.get(index).markAsDone();
+                System.out.println("Nice! I've marked this task as done:");
+                System.out.println("  " + tasks.get(index).toString());
+                saveTasksToFile();
+            } else {
+                throw new BebeException("Invalid task number.");
+            }
+        } catch (NumberFormatException e) {
+            throw new BebeException("Please provide a valid task number.");
+        }
+    }
+
+    /**
+     * Marks a task as not done.
+     *
+     * @param taskNumber The index of the task to mark as not done.
+     */
+    private static void markTaskAsNotDone(String taskNumber) throws BebeException {
+        try {
+            int index = Integer.parseInt(taskNumber) - 1;
+            if (index >= 0 && index < tasks.size()) {
+                tasks.get(index).markAsNotDone();
+                System.out.println("OK, I've marked this task as not done yet:");
+                System.out.println("  " + tasks.get(index).toString());
+                saveTasksToFile();
+            } else {
+                throw new BebeException("Invalid task number.");
+            }
+        } catch (NumberFormatException e) {
+            throw new BebeException("Please provide a valid task number.");
+        }
+    }
+
+    /**
+     * Saves the current list of tasks to a file.
+     */
+    private static void saveTasksToFile() {
+        try {
+            File directory = new File("./data");
+            if (!directory.exists()) {
+                directory.mkdir(); // Create directory if it doesn't exist
+            }
+
+            FileWriter writer = new FileWriter(FILE_PATH);
+            for (Task task : tasks) {
+                writer.write(taskToFileFormat(task) + "\n");
+            }
+            writer.close();
+        } catch (IOException e) {
+            System.out.println("Error saving tasks: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Converts a Task object into a string for saving in a file.
+     */
+    private static String taskToFileFormat(Task task) {
+        String type = "";
+        String status = task.isDone ? "1" : "0"; // 1 for done, 0 for not done
+
+        if (task instanceof Todo) {
+            type = "T";
```

Comment:
Maybe the magic literial `T` could be refactored? Similar case for several other magic literals